### PR TITLE
Add codelens for in-editor run button

### PR DESCRIPTION
**Pull request by sdankel**
This adds a Run button above every runnable function (defined by the LSP server). When clicked, it runs `forc run` on the file.

In the future, we can distinguish between main functions and test functions, which could also be runnable.

Aug-01-2022 17-43-13

Fixes https://github.com/FuelLabs/sway-vscode-plugin/issues/54

Related https://github.com/FuelLabs/sway/pull/2441

### CODE REVIEW

File: client/src/client.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  LanguageClientOptions,
+  LSPErrorCodes,
+  RequestType,
+  ServerOptions,
+} from 'vscode-languageclient/node';
+import { log, sleep } from './util';
+
+let client: LanguageClient | null;
+
+export const createClient = (
```
Comment (sdankel) [inline client/src/client.ts:13]: This is just a singleton instance of the client that makes it easily accessible.

File: client/src/main.ts [hunk 2]
```diff
@@ -7,52 +6,74 @@ import { Program, Function, ProgramProvider } from './program';
...
   // Register tree views
   const rootPath =
-    vscode.workspace.workspaceFolders &&
-    vscode.workspace.workspaceFolders.length > 0
-      ? vscode.workspace.workspaceFolders[0].uri.fsPath
+    workspace.workspaceFolders && workspace.workspaceFolders.length > 0
+      ? workspace.workspaceFolders[0].uri.fsPath
       : undefined;
-  const contractProvider = vscode.window.registerTreeDataProvider(
+  const contractProvider = window.registerTreeDataProvider(
```
Comment (sdankel) [inline client/src/main.ts:45]: Fix the imports :)

**Review by JoshuaBatty — APPROVED** [review]
Nice, looks great.

**Comment by JoshuaBatty** [comment]
this PR along with [https://github.com/FuelLabs/sway/pull/2441](https://github.com/FuelLabs/sway/pull/2441) closes #74

**Pull request merged**